osg-pilot-container/osgpilot_meter

```diff
@@ -440,6 +459,30 @@ def detect_new_jobs(sqldb, current_ts):
     DebugPrint(INFO, "- deleted %s rows from inbox" % rc2)
 
 
+def detect_and_update_partial_jobs(sqldb):
+    DebugPrint(INFO, "Reporting and breaking up partial jobs")
+
+    insert_sql = """
+        insert into outbox (%s)
+        select *
+          from cutoff
+    """ % ", ".join(SEND_ATTRS)
+
+    # reset aggregates
+    update_sql = """
+        update jobs
+           set DaemonStartTime = last_updated,
+               total_cpu = 0,
+               MemoryUsage = 0
+         where last_updated - DaemonStartTime >= {partial_record_cutoff};
+    """.format(partial_record_cutoff=PARTIAL_RECORD_CUTOFF)
+
+    rc1 = sqldb.execute(insert_sql, (dead_cutoff,)).rowcount
+    rc2 = sqldb.execute(update_sql, (dead_cutoff,)).rowcount
```

Comment:
What happens if the job makes the cutoff for one query but not the other (suppose that the OS sleeps the process for 2 hours between these two SQL statements)?  Do we just loose the hours?

Reply:
The timestamp for all incoming records is the same in the inbox table, then aggregated into the jobs table.  Both queries use the same `last_updated - DaemonStartTime >= {partial_record_cutoff}` condition from the jobs table, not the current query execution timestamp.  So, adding a 2 hour sleep between those two execute statements (or anywhere else, for that matter) should not affect the results.